packages/jquery-odata-client/src/JQueryODataClient.ts

```diff
@@ -50,15 +50,10 @@ export class JQueryODataClient implements ODataClient<AjaxRequestConfig> {
         },
         error: (jqXHR: JQuery.jqXHR, textStatus: string, thrownError: string) => {
           const message = this.getErrorMessage(jqXHR.responseJSON);
-
-          reject(
-            Object.assign(new Error(), {
-              status: jqXHR.status || 0,
-              code: jqXHR.statusText,
-              data: jqXHR.responseJSON,
-              message,
-            } as RequestError)
-          );
+          if (message) {
+            reject(new JQueryODataClientError("OData server error: " + message, { cause: jqXHR }));
```

Comment:
either a  `return` or an `else` block => makes it clearer I think. At least I am thinking now, if this works or if `reject` gets called twice now....